### PR TITLE
feat: cli to display ipcache by labels

### LINKED ISSUE #34632: CFP: Provide `cilium ip get` command enhancements to filter output by CIDR or labels.
## Cilium Feature Proposal
At present, there is no option to display information about the IP cache (ipcache) by CIDR or labels. This necessitates capturing all ipcache information using the "cilium ip list" command and parsing it subsequently. The suggested feature aims to display ipcache information based on CIDR or labels.

Once implemented, users can utilize the `cilium ip` command as follows:
```
Usage:
  cilium ip get ( <cidr> |-l <identity labels> ) [flags]

Flags: -h, --help             help for get
  -l, --labels strings   list of labels
  -o, --output string    json| yaml| jsonpath='{}'
  -v, --verbose          Print all fields of ipcache
```
**Is your proposed feature related to a problem?**
No

**Describe the feature you'd like**

Include any specific requirements you need

**(Optional) Describe your proposed solution**

I will contribute the code to provide above CLI.

### PR DESCRIPTION
**Pull request by vasu-dasari**
Added new CLI to display ipcache entries by cidr and labels.

Display IP Cache information
```
Usage:
cilium-dbg ip get ( <cidr> |-l <identity labels> ) [flags]

Flags: -h, --help help for get
-l, --labels strings list of labels
-o, --output string json| yaml| jsonpath='{}'
-v, --verbose Print all fields of ipcache
```
Fixes: #34632
Signed-off-by: Vasu Dasari <vasudasari@google.com>

```release-note
introducing a new CLI option to display ipcache information by labels or cidr
```

### CODE REVIEW

**Comment by vasu-dasari** [comment]
CLI display from my testing:
Help text:
```
root@gke-vasudasari-cilium-1-default-pool-4a9bc11f-kzys:/home/cilium# cilium-dbg ip get -h
Display IP Cache information

Usage:
  cilium-dbg ip get ( <cidr> |-l <identity labels> ) [flags]

Flags:
  -h, --help             help for get
  -l, --labels strings   list of labels
  -o, --output string    json| yaml| jsonpath='{}'
  -v, --verbose          Print all fields of ipcache

Global Flags:
      --config string   Config file (default is $HOME/.cilium.yaml)
  -D, --debug           Enable debug messages
  -H, --host string     URI to server-side API
```
Query by CIDR:
```
root@gke-vasudasari-cilium-1-default-pool-4a9bc11f-kzys:/home/cilium# cilium-dbg ip get 10.168.0.12/32
IP               IDENTITY        SOURCE
10.168.0.12/32   reserved:host
```
```
root@gke-vasudasari-cilium-1-default-pool-4a9bc11f-kzys:/home/cilium# cilium-dbg ip get -l k8s:io.kubernetes.pod.namespace=default
===========================
Labels:
    k8s:io.kubernetes.pod.namespace=default
===========================
IP                HOST          IDENTITY   POD                           NAMESPACE
10.80.0.2/32      10.168.0.13   76103      rebel-base-55756c47d5-kzdsr   default
10.80.0.99/32     10.168.0.13   74181      x-wing-6bb767fcb8-757wc       default
10.80.3.4/32      10.168.0.12   74181      x-wing-6bb767fcb8-4jwsk       default
10.80.3.203/32    10.168.0.12   76103      rebel-base-55756c47d5-gjhm5   default
10.100.1.23/32    10.168.0.16   183733     rebel-base-55756c47d5-f4rv6   default
10.100.1.54/32    10.168.0.16   157998     x-wing-6bb767fcb8-2whv5       default
10.100.3.140/32   10.168.0.15   157998     x-wing-6bb767fcb8-947rj       default
10.100.3.206/32   10.168.0.15   183733     rebel-base-55756c47d5-sf8c5   default
```

**Comment by nathanjsweet** [comment]
@vasu-dasari Please sign-off your first commit and remove the last empty commit. Thanks :slightly_smiling_face: !

**Review by nathanjsweet — APPROVED** [review]

**Comment by nathanjsweet** [comment]
@vasu-dasari Pinging you on getting this merged. Please [sign-off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) your first commit and remove the empty last commit from your PR. Thanks! :slightly_smiling_face:

File: cilium-dbg/cmd/ip_get.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import (
+	"fmt"
+	"net"
+	"os"
+	"sort"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+
+	ipApi "github.com/cilium/cilium/api/v1/client/policy"
```
Comment (doniacld) [inline cilium-dbg/cmd/ip_get.go]: nit: can you remove the capital in the alias for `ipapi` (cf https://go.dev/blog/package-names)

**Review by doniacld — CHANGES_REQUESTED** [review]
One nit otherwise lgtm!

**Comment by derailed** [comment]
@vasu-dasari Could you take another look when you get a chance? Thank you!

File: cilium-dbg/cmd/ip_get.go (hunk 1 above)
Comment (vasu-dasari) [reply cilium-dbg/cmd/ip_get.go]: Yes. Made the changes as suggested. Thanks for reviewing.

**Review by vasu-dasari — COMMENTED** [review]
Added `Signed-off-by` to my commit. And addressed comment from @doniacld

**Review by doniacld — APPROVED** [review]
LGTM! Thanks for the change

**Review by jrajahalme — APPROVED** [review]

**Pull request merged**